### ISSUE TITLE
[Feature Request] GUI-applet for Wicked Network Framework

### ISSUE THREAD

**Issue by MikhailKasimov** [openSUSE/wicked#643]
Hello! There is an idea about GUI-applet for Wicked Network Framework. For example, OpenSUSE 13.2/Leap uses Wicked Network Framework as the network configuration tool by default. But desktops (e.g. XFCE) have only gnome-nm-applet, which is acceptable only if user chose NetworkManager to manage the network. See attached file. Messages are in Russian, so I translate -- "Система" -- "Сетевые соединения" (System -- Network Connections). But this functionality is meaningless while using Wicked and\or other network management framework. Ok, I can file feature requests on KDE, GNOME, XFCE, ..., bla-bla other desktops bug-trackers and wait for very-very-...-very long time while they consider them. But it seems to be more logically to file this feature request here (even not on fate.opensuse.org, which seems to be a good place to ignore user's ideas in most of them, but it's my personal impression only) to have Qt-based and Gtk-based GUI-applet for Wicked Network Framework from only source, which can be built later in distro-specific desktops assemblies. Thank you! ![opensuse_gnome_nm_applet](https://cloud.githubusercontent.com/assets/7167300/14643687/05709e32-0658-11e6-9917-e622c61adcaf.png)

**Comment by mtomaschewski** [comment]
Yes, the idea to implement one is old, but currently there is nobody willing to implement one.

**Comment by MikhailKasimov** [comment]
The idea is when (open)SUSE is being installed to choose what network manager is going to be in use. But I don't remember if (openSUSE) installer asks about that. It installs Wicked active by default and NM as an alternative, which can be switched on manually later instead of Wicked. But anyway it installs gnome-nm-applet by default and that is illogically --  to install applet for alternative, which is off by default. So, i see two way for now: 1) to have separated ```wicked-gui-applet-gtk``` and ```wicked-gui-applet-qt``` RPMs, which can be installed manually or 2) to have ```wicked-gui-applet-gtk``` and ```wicked-gui-applet-qt``` RPMs in main Wicked package dependencies initially.

``` but currently there is nobody willing to implement one. ```

And what are their arguments?

**Comment by OneAdder** [comment]
Hello. Any progress?

**Comment by patogit** [comment]
Thinking about why this proposal might make sense (+) or not (-) :
- it would take time to create. 
- it would even more time to maintain over the coming years. 
- wicked has been used without a GUI so far (over 7 years?), and is mostly used by people who accept or even prefer text files and command line (not a GUI), and people who want a GUI can switch to NetworkManager easily, so a GUI for wicked isn't necessary. 
- wicked is capable of more complex arrangements than NetworkManager (?) thus requires a more complex GUI.
+ a GUI would make the benefits of wicked more accessible to users who prefer a GUI, especially common desktop users.
+ more widespread use of wicked might bring more attention to it, and result in more people contributing.

Do these items seem true? What corrections can we make to this list? 

I figure if we can think about why this would make sense or not, we can think of a practical way forward... Maybe start with a very basic GUI that only implements a few configurations?

**Comment by mtomaschewski** [comment]
> Maybe YaST is already functioning as a GUI for wicked?
> 
> I click around and see that this does indeed appear to be a GUI for wicked.

Yes, yast2-network configures wicked, but it is not a GUI applet Mikhail has in his mind.
I'd keep prefer to put it into a separate `wicked-gui` or similar project, but it would still
require a fate.opensuse.org / jira.suse.com feature request to allocate GUI developer
resources, which we don't have here. I'll keep the issue open, but it's unlikely that we
will find the time to implement an applet. We will try to make some improvements to be
able to push user (!root) config to nanny  what is  currently not code complete and
thus disabled, but this is also an _when the time permits_ enhancement effort only.

On SLES it's wicked, on SLED it's NetworkManager, but the openSUSE installer shows
an overview about the installation proposal and also if wicked or NetworkManager will
be used with 1-Click option to switch -- the default setting depends on the distribution
pattern choice [Gnome/KDE/X11 Desktop, Sever, ...] made one step before.
